docs/csharp/programming-guide/concepts/linq/snippets/LinqExtensions.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace linq
+{
+    // <LinqExtensionClass>
+    public static class LINQExtension
+    {
+        public static double Median(this IEnumerable<double>? source)
+        {
+            var countOfElementsInTheSet = source?.Any() ?? false;
+
+            if (!countOfElementsInTheSet)
+            {
+                throw new InvalidOperationException("Cannot compute median for a null or empty set.");
+            }
+
+            var sortedList = (from number in source
+                              orderby number
+                              select number).ToList();
+
+            int itemIndex = sortedList.Count / 2;
+
+            if (sortedList.Count % 2 == 0)
+            {
+                // Even number of items.
+                return (sortedList[itemIndex] + sortedList[itemIndex - 1]) / 2;
+            }
+            else
+            {
+                // Odd number of items.
+                return sortedList[itemIndex];
+            }
+        }
+    }
+    // </LinqExtensionClass>
+
+    public static class OtherExtensions
+    {
+        // <IntOverload>
+        //int overload
+        public static double Median(this IEnumerable<int> source) =>
+            (from num in source select (double)num).Median();
+        // </IntOverload>
+
+        // <GenericOverload>
+        // Generic overload.
+        public static double Median<T>(this IEnumerable<T> numbers,
+                               Func<T, double> selector) =>
+            (from num in numbers select selector(num)).Median();
+        // </GenericOverload>
+
+        // <SequenceElement>
+        // Extension method for the IEnumerable<T> interface.
+        // The method returns every other element of a sequence.
+        public static IEnumerable<T> AlternateElements<T>(this IEnumerable<T> source)
+        {
+            int i = 0;
+            foreach (var element in source)
+            {
+                if (i % 2 == 0)
+                {
+                    yield return element;
+                }
+                i++;
+            }
```

Comment:
Would it be better to convert this foreach to for loop, and increase the iteration variable by 2 each time?
That will reduce the number of iterations to the half, and eliminate the if condition.

Or I'm missing something?

Reply:
It's a `foreach` loop because `IEnumerable<T>` doesn't support an indexer, or a `Count` property. So, to make that change, the parameter would need to be a `ICollection<T>` or some collection that had indexers.